File: 349.IntersectionOfTwoArrays.cs
using System;
using System.Collections.Generic;

namespace IntersectionOfTwoArrays;
public class Solution {
  public int[] Intersection(int[] nums1, int[] nums2) {
    HashSet > set1 = new HashSet<int>(nums1);
    HashSet<int> set2 = new HashSet<int>(nums2);
    set1.IntersectWith(set2);
    return set1.ToArray();
  }
}

File: Solutions/HashTable/349.IntersectionofTwoArrays.py
from typing import List


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1 = set(nums1)
        set2 = set(nums2)
        # set is O1 average time complexity for lookups
        result = []
        for num in set1:
            if num in set2:
                result.append(num)
        return result
